canonical-web-and-design/maas-ui
# feat(ui): move delete KVM action to settings tab

## Done

- Moved delete KVM action to settings tab
- Removed Take action menu for LXD KVMs since now all actions can be performed in the tabs
- Added `decompose` boolean to delete form for LXD KVMs
- Renamed `PodConfiguration` to `KVMConfiguration` since we no longer support rsds

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Add a LXD KVM. You can do this by using the same address as `bolla` and selecting an existing project.
- Check that there is no "Take action" menu in the header
- Compose a machine (values are unimportant, just use defaults)
- Go to the Settings tab and click "Remove KVM" in the Danger zone and check that the Delete form opens
- Remove the KVM and check that you are taken back to the KVM list
- Re-add the exact same LXD KVM (you can use a different name, but the address and project must be the same)
- Check that the VM that you composed before shows up in the VM list automatically
- Now open the Delete form and check the checkbox to delete all VMs before submitting
- Re-add the same LXD KVM one more time and check that the VM does not show up in the VM list
- Go to the details page of a virsh KVM and check that the Take action menu and Delete form have not changed

## Fixes

Fixes #2096
